src/v/storage/segment_index.cc

```diff
@@ -186,7 +192,7 @@ ss::future<> segment_index::flush() {
       .then(
         [this] { return ss::make_file_output_stream(ss::file(_out.dup())); })
       .then([this](ss::output_stream<char> out) {
-          auto b = _state.checksum_and_serialize();
+          auto b = serde::to_iobuf(_state.copy());
```

Comment:
@felixguendling another case where avoiding a copy would be useful. in this case `to_iobuf` isn't a sink for `_state`.